pkgs/by-name/su/sui/package.nix

```
@@ -0,0 +1,64 @@
+# export a function that can build various Linux Sui Versions
+{
+  lib,
+  system ? stdenv.hostPlatform.system,
+  stdenv,
+  fetchurl,
+  autoPatchelfHook,
+  gccForLibs,
+  versionCheckHook,
+}:
+
+let
+  arch =
+    if "${system}" == "aarch64-linux" then
+      "aarch64"
+    else if "${system}" == "x86_64-linux" then
+      "x86_64"
+    else
+      throw "using unsupported system"; # architecture check for builder
+
+  mk_release =
+    # builder function
+    {
+      type ? "mainnet",
+      version_number,
+      arch,
+      vhash,
+    }:
+
+    stdenv.mkDerivation (finalAttrs: {
+      pname = "sui-patched";
+      description = "patched Sui binaries for nix compat";
+      version = "${type}-v${version_number}";
+      src = fetchurl {
+        url = "https://github.com/MystenLabs/sui/releases/download/${finalAttrs.version}/sui-${finalAttrs.version}-ubuntu-${arch}.tgz";
+        hash = vhash;
+      };
+      sourceRoot = ".";
+      nativeBuildInputs = [ autoPatchelfHook ];
+      buildInputs = [ gccForLibs.lib ];
+      installPhase = ''
+        runHook preInstall
+        mkdir -p $out/bin
+        install -m 755 sui* move-analyzer $out/bin
```

https://nixos.org/manual/nixpkgs/unstable/#installshellfiles-installbin
```suggestion
        installBin sui* move-analyzer $out/bin
```

---

Thanks for all the advice and the comments I'll make some changes today and put it all together.

---

I look at the source code for installBin its a neat function but in this case I think it  becomes necessary for more complicated builds.
Its ultimately just a helper function, but this build is so simple I dont think it requires adding `installShellFiles` as a nativeBuildInput.